pkgs/os-specific/linux/zfs/default.nix

```diff
@@ -226,17 +226,17 @@ in {
 
   zfsUnstable = common {
     # check the release notes for compatible kernels
-    kernelCompatible = kernel.kernelAtLeast "3.10" && kernel.kernelOlder "5.15";
+    kernelCompatible = kernel.kernelAtLeast "3.10" && kernel.kernelOlder "5.16";
     latestCompatibleLinuxPackages = linuxPackages_5_14;
 
     # this package should point to a version / git revision compatible with the latest kernel release
     # IMPORTANT: Always use a tagged release candidate or commits from the
     # zfs-<version>-staging branch, because this is tested by the OpenZFS
     # maintainers.
-    version = "2.1.1";
-    # rev = "0000000000000000000000000000000000000000";
+    version = "2021-11-04";
+    rev = "4b87c1981d47655fd2eec2fb95179818240c4945";
```

Comment:
Please use a commit from the `zfs-2.1.2-staging` branch, *not* from the `master` branch. The `master` branch isn't tested and we had problems with this before.

Reply:
@zhaofengli can say more about this.

Reply:
See also https://github.com/NixOS/nixpkgs/pull/136340

Reply:
Ok. I was not aware of staging branches.

Reply:
Hm, the `META` file still only states 5.14 (https://github.com/openzfs/zfs/blob/zfs-2.1.1/META#L9). Instead of “unlocking” it for everyone and potentially corrupting everyone's filesystem, could you use `allowBroken = true;`?